docker/Dockerfile

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2020 Intel Corporation. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#   * Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#   * Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in
+#     the documentation and/or other materials provided with the
+#     distribution.
+#   * Neither the name of Intel Corporation nor the names of its
+#     contributors may be used to endorse or promote products derived
+#     from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+
+FROM ubuntu:18.04 as builder
+
+RUN apt-get update --fix-missing
+RUN apt-get install -y build-essential ocaml ocamlbuild automake autoconf libtool wget python libssl-dev 
+RUN apt-get install -y git libcurl4-openssl-dev protobuf-compiler libprotobuf-dev debhelper cmake lsb-release
```

Comment:
I was only able to build this image after adding `curl` as a dependency package here.

Reply:
hmm, can you tell which step did it fail or the error message?

Reply:
This is the step and error message obtained:

```bash
Step 12/21 : RUN make psw_install_pkg
 ---> Running in 44272a018f53
fatal: not a git repository (or any of the parent directories): .git
curl --output dcap_source.tar.gz -L --tlsv1 https://github.com/intel/SGXDataCenterAttestationPrimitives/archive/DCAP_1.4.tar.gz
make: curl: Command not found
Makefile:62: recipe for target 'dcap_source' failed
make: *** [dcap_source] Error 127
The command '/bin/sh -c make psw_install_pkg' returned a non-zero code: 2
```

Steps to reproduce error:

```bash
git clone https://github.com/haitaohuang/linux-sgx.git -b docker_pr3
cd linux-sgx/docker
docker build --target aesm -t sgx_aesm -f ./Dockerfile ../
```

Reply:
Ah, thanks
I used git clone --recurse-submodules.